varintf/varintf_test.go

```diff
@@ -95,6 +96,34 @@ func TestFactomSpecExamples(t *testing.T) {
 	}
 }
 
+// TestVarInt_Panic ensures the varint decoding will never panic from invalid data
+func TestVarInt_Panic(t *testing.T) {
+	assert := assert.New(t)
+	// a quick helper func to test these things
+	testDecode := func(expA uint64, expR int64, data []byte, msg string) {
+		a, r := Decode([]byte{179, 144})
+		assert.Equal(expA, a, msg)
+		assert.Equal(expR, r, msg)
+	}
+	t.Run("known invalid slices", func(t *testing.T) {
+		testDecode(0, -1, []byte{179, 144}, "not enough bytes")
+	})
+
+	// The easiest way to search for panics is with random data
+	t.Run("test error handling (no panic)", func(t *testing.T) {
+		for i := 0; i < 1000; i++ {
+			d := make([]byte, rand.Intn(20))
+			rand.Read(d)
```

Comment:
I believe the seed should be explicitly set somewhere to a constant value so that "random" tests are reproducible. Or is the seed always the same if not otherwise initialized?

Reply:
The seed is the same each time you run unless someone explicitly changes the seed. But the seed is global, so there is some value is initializing it upfront. Since even adding a unit test, or changing the execution order would change the values.

I would argue that those subtle changes though are fine. As long as the tests are reproducible, given you know the test conditions (`go test -run TestXYZ` vs `go test ./...`), I think that is sufficient.